protocol/x/clob/keeper/liquidations.go

```diff
@@ -105,7 +106,18 @@ func (k Keeper) LiquidateSubaccountsAgainstOrderbook(
 			}
 
 			// Return unexpected errors.
-			return err
+			return 0, err
+		}
+
+		// Skip liquidation if the ClobPair for the liquidation order is in final settlement. There is a separate flow
```

Comment:
We skip attempting to place the liquidation if the market is in final settlement, as liquidation matches are not allowed in closed markets. We do append the order to unfilledLiquidations so that the deleveraging flow may naturally pick up negative TNC subaccounts in the market and deleverage them as part of the standard deleveraging flow

Reply:
Why not pick another perpetual ID to liquidate?

What if the subaccount is liquidatable but has multiple positions? Closing their open but final settled position at the oracle price won't change their collateralization, so IMO could make sense to just liquidate another position

However I could see us leaning towards the simplest approach and just waiting another block since I _think_ it picks a random position to liquidate

Wdyt?

Reply:
yeah admittedly was going for simplicity here. Picking a new position to liquidate would mean introducing a new flow here where we selectively re-generate liquidation orders depending on clob pair status and I just don't think that is worth it for now. If you feel strongly, we could implement this later but for now I think this is fine